setup.py

```diff
@@ -221,7 +223,33 @@ def get_extensions():
         except ImportError:
             pass
 
-        if is_rocm_pytorch or torch.cuda.is_available() or os.getenv(
+        if os.getenv('MMCV_WITH_DIOPI', '0') == '1':
+            import mmengine  # NOQA: F401
+            major, minor, micro = mmengine.version_info
+            if major == 0 and (minor < 7 or (minor == 7 and micro < 4)):
+                raise RuntimeError(f'mmengine >= 0.7.4 is required but \
+                        {mmengine.__version__} is installed')
```

Comment:
We can use the function `digit_version` in
https://github.com/open-mmlab/mmengine/blob/a2e410bd48890c54f16c527d67c371ed8562facd/mmengine/utils/version_utils.py#L9